core/domain/feedback_services.py

```diff
@@ -172,3 +188,99 @@ def get_thread_analytics(exploration_id):
+    """Creates a new SuggestionModel object and the corresponding
+    FeedbackThreadModel object."""
+
+    thread_id = _create_models_for_thread_and_first_message(
+        exploration_id, state_name, author_id, 
+        DEFAULT_SUGGESTION_THREAD_SUBJECT, EMPTY_TEXT, True)
+    (feedback_models.SuggestionModel.create(
+        exploration_id, thread_id, author_id, exploration_version, state_name,
+        suggestion_content))
+
+
+def get_suggestion(exploration_id, thread_id):
+    return feedback_models.SuggestionModel.get_by_exploration_and_thread_id(
+        exploration_id, thread_id)
+
+
+def get_open_threads(exploration_id, suggestions):
+    """If suggestions is True, return a list of all open threads that have a
+    suggestion, otherwise return a list of all open threads that do not have a 
+    suggestion."""     
+
+    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    open_threads = []
+    for thread in threads:
+        if (bool(thread.has_suggestion) == bool(suggestions) and 
+            thread.status == feedback_models.STATUS_CHOICES_OPEN):
+            open_threads.append(thread)
+    return open_threads
+
+
+def get_closed_threads(exploration_id, suggestions):
+    """If suggestions is True, return a list of all closed threads that have a
+    suggestion, otherwise return a list of all closed threads that do not have a 
+    suggestion."""     
+
+    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    closed_threads = []
+    for thread in threads:
+        if (bool(thread.has_suggestion) == bool(suggestions) and 
+            thread.status != feedback_models.STATUS_CHOICES_OPEN):
+            closed_threads.append(thread)
+    return closed_threads
+
+
+def get_all_suggestion_threads(exploration_id):
+    """Return a list of all threads with suggestions."""
+
+    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    all_threads = []
+    for thread in threads:
+        if thread.has_suggestion:
+            all_threads.append(thread)
+    return all_threads
+
+	
+def _is_suggestion_valid(thread_id, exploration_id):
+    """Check if the suggestion is still valid. A suggestion is considered
+    invalid if the exploration version has updated since the suggestion was
+    made."""
+
+    return (feedback_models.SuggestionModel.get_by_exploration_and_thread_id(
+        exploration_id, thread_id).exploration_version ==
+        exp_models.ExplorationModel.get(id=exploration_id).version)
+
+
+def accept_suggestion(editor_id, change_list, thread_id, exploration_id, 
+                      message):
+    """If the suggestion is valid, accepts it by updating the exploration.
+    Raises an exception if the exploration is not valid."""
+
+    from core.domain import exp_services
+
+    if _is_suggestion_valid(thread_id, exploration_id):
+        commit_message = ('Accepted suggestion by %s: %s' % (
+            editor_id, message)) 
```

Comment:
Using human readable username now.(We should start doing the same in editor.py as well, when we call update_exploration)